src/engine/enginemaster.h

```diff
@@ -126,68 +128,104 @@ class EngineMaster : public QObject, public AudioSource {
+        bool m_fadeout;
     };
 
     class GainCalculator {
       public:
         virtual double getGain(ChannelInfo* pChannelInfo) const = 0;
     };
-    class ConstantGainCalculator : public GainCalculator {
+    class PflGainCalculator : public GainCalculator {
       public:
         inline double getGain(ChannelInfo* pChannelInfo) const {
-            return pChannelInfo->m_pChannel->isTalkover() ? m_dTalkoverGain : m_dGain;
+            Q_UNUSED(pChannelInfo);
+            return m_dGain;
         }
         inline void setGain(double dGain) {
             m_dGain = dGain;
         }
-        inline void setTalkoverGain(double dGain) {
-            m_dTalkoverGain = dGain;
-        }
       private:
         double m_dGain;
-        double m_dTalkoverGain;
+    };
+    class TalkoverGainCalculator : public GainCalculator {
+      public:
+        inline double getGain(ChannelInfo* pChannelInfo) const {
+            Q_UNUSED(pChannelInfo);
+            return 1.0;
+        }
     };
     class OrientationVolumeGainCalculator : public GainCalculator {
       public:
         OrientationVolumeGainCalculator()
-                : m_dVolume(1.0), m_dLeftGain(1.0), m_dCenterGain(1.0), m_dRightGain(1.0),
-                  m_dTalkoverGain(1.0) { }
+                : m_dVolume(1.0),
+                  m_dLeftGain(1.0),
+                  m_dCenterGain(1.0),
+                  m_dRightGain(1.0) {
+        }
 
         inline double getGain(ChannelInfo* pChannelInfo) const {
-            if (pChannelInfo->m_pMuteControl->get() > 0.0) {
-                return 0.0;
-            }
-            if (pChannelInfo->m_pChannel->isTalkover()) {
-                return m_dTalkoverGain;
-            }
             const double channelVolume = pChannelInfo->m_pVolumeControl->get();
             const double orientationGain = EngineMaster::gainForOrientation(
-                pChannelInfo->m_pChannel->getOrientation(),
-                m_dLeftGain, m_dCenterGain, m_dRightGain);
+                    pChannelInfo->m_pChannel->getOrientation(),
+                    m_dLeftGain, m_dCenterGain, m_dRightGain);
             return m_dVolume * channelVolume * orientationGain;
         }
 
-        inline void setGains(double dVolume, double leftGain, double centerGain, double rightGain,
-                             double talkoverGain) {
+        inline void setGains(double dVolume, double leftGain,
+                double centerGain, double rightGain) {
             m_dVolume = dVolume;
             m_dLeftGain = leftGain;
             m_dCenterGain = centerGain;
             m_dRightGain = rightGain;
-            m_dTalkoverGain = talkoverGain;
         }
 
       private:
-        double m_dVolume, m_dLeftGain, m_dCenterGain, m_dRightGain, m_dTalkoverGain;
+        double m_dVolume;
+        double m_dLeftGain;
+        double m_dCenterGain;
+        double m_dRightGain;
+    };
+    template<typename T, unsigned int CAPACITY>
+    class FastVector {
```

Comment:
> I have actually started to introduce QVarLengthArray, but it feels stupid to code around the lenght check like ->data()[i], which will reintroduce the same problems with complex types you have pointed out.

```->data()[i]``` instead of accessors doesn't have any complex type problems -- trying to replace ```append``` or ```replace``` would though.